src/features/addonList/AddOnList.test.tsx

```diff
@@ -7,65 +7,55 @@
  */
 
 import { describe, it, expect } from "vitest";
-/*import type { Addon } from "../../types/AddOnTypes";
-import { AddonCategory } from "../../types/AddOnTypes";
-import { Author } from "../../types/AuthorTypes";
-import { User } from "../../types/UserTypes";
 import { renderWithProviders } from "../../utils/test-utils";
-import AddOnList from "./AddOnList";*/
+import AddOnList from "./AddOnList";
+import { addonList } from "../../temp/tempAddons";
+import { server } from "../../setupTests";
+import { HttpResponse, http } from "msw";
 
-// Sample add-on data
-/*const mockAddOns: Addon[] = [
-  {
-    id: "1",
-    name: "Mock Addon 1",
-    summary: "Mock Summary 1",
-    category: AddonCategory.VISUALISATION
-  },
-  {
-    id: "2",
-    name: "Mock Addon 2",
-    summary: "Mock Summary 2",
-    category: AddonCategory.VISUALISATION
-  }
-];*/
-
-// TODO: Doesn't succeed because of loading screen when fetching from backend
 describe("AddOnList component", () => {
-  it("temp", () => {
-    expect(true).toEqual(true);
-  });
-  /*it("renders AddOnCard components for all add-ons", async () => {
-    const mockState: AddOnListState = {
-        allAddOns: mockAddOns,
-        searchTerm: "",
-        status: "succeeded",
-        error: null
-    };
+  it("renders AddOnCard components for all add-ons", async () => {
+    const { findAllByTestId, findByText } = renderWithProviders(<AddOnList />);
 
-    const { getAllByTestId, store } = renderWithProviders(<AddOnList />, {
-      preloadedState: { addons: mockState }
-    });
+    expect(findByText("Loading...")).toBeDefined();
 
-    // Expect two AddOnCard components to be rendered
-    const addOnCards = getAllByTestId("addon-card");
-    expect(addOnCards.length).toBe(2);
+    const addOnCards = await findAllByTestId("addon-card");
+    expect(addOnCards.length).toBe(addonList.length);
   });
 
   it("filters add-ons based on searchTerm", async () => {
-    const mockState: AddOnListState = {
-        allAddOns: mockAddOns,
-        searchTerm: "mock addon 1",
-        status: "succeeded",
-        error: null
-    };
-
-    const { getAllByTestId } = renderWithProviders(<AddOnList />, {
-      preloadedState: {addons: mockState}
+    const { findAllByTestId, getByText } = renderWithProviders(<AddOnList />, {
+      preloadedState: { addons: { searchTerm: addonList[0].name } }
     });
 
     // Expect only one AddOnCard (the one matching the search term)
-    const addOnCards = getAllByTestId("addon-card");
+    const addOnCards = await findAllByTestId("addon-card");
+
     expect(addOnCards.length).toBe(1);
-  });*/
+    expect(getByText(addonList[0].summary)).toBeDefined();
+  });
+
+  it("displays message when no addons are found with search term", async () => {
+    const { findByText } = renderWithProviders(<AddOnList />, {
+      preloadedState: { addons: { searchTerm: "qwerty" } }
+    });
+
+    const message = await findByText(
+      "No Add-ons found with the given search term"
```

Comment:
Dit lijkt me testen van een implementation detail. Wat als we de tekst willen aanpassen?